at-dungnguyen/Final/main.rb

```diff
@@ -0,0 +1,399 @@
+    p '0. Exit'
+  end
+
+  def execute_leave(function)
+    case function
+    when 1
+      send_request
+    when 2
+      update_request
+    when 3
+      delele_request(@user)
+    when 4
+      list_approved
+    when 5
+      @flag_leave = false
+    when 6
+      @flag_leave = false
+      @user = nil
+    when 7
+      list_need_approve if @user['role'] == SessionController::LEADER
+    when 8
+      update_status if @user['role'] == SessionController::LEADER
+    when 0
+      @flag_leave = false
+      execute2(0)
+    end
+  end
+
+  def send_request
+    p 'Reason'
+    reason = gets.chomp
+    p 'Total date off'
+    total_date_off = input
+    time = Time.now
+    request = Leave.new(@user['id'], reason, total_date_off, 0, time.strftime('%F %T'))
+    clear
+    if @lc.add_item(request)
+      p 'Send Sucess'
+    else
+      p 'Send Error'
+    end
+  end
+
+  def update_request
+    p 'ID leave request'
+    id = input
+    p 'Reason'
+    reason = gets.chomp
+    p 'Total date off'
+    total_date_off = input
+    leave = Leave.new(@user['id'], reason, total_date_off, 0, 0)
+    if @lc.update_leave(id, leave, @user)
+      p 'Update Sucess'
+    else
+      p 'Update Error'
+    end
+  end
+
+  def delele_request(user)
+    p 'ID leave request'
+    id = input
+    if @lc.del_item(id, user)
+      p 'Delete Sucess'
+    else
+      p 'Delete Error'
+    end
+  end
+
+  def list_approved
+    if @lc.all_approved.any?
+      @lc.all_approved.each do |leave|
+        p "ID: #{leave['id']} - User: #{leave['id_user']} - Reason: #{leave['reason']} - Date off: #{leave['total_date_off']}"
+      end
+    else
+      p 'No result'
+    end
+  end
+
+  def list_need_approve
+    if @lc.need_approve.any?
+      @lc.need_approve.each do |leave|
+        p "ID: #{leave['id']} - User: #{leave['id_user']} - Reason: #{leave['reason']} - Date off: #{leave['total_date_off']}"
+      end
+    else
+      p 'No result'
+    end
+  end
+
+  def update_status
+    p 'ID leave request'
+    id = input
+    p 'Approve or Reject'
+    status = input
+    if @lc.update_status(id, status)
+      p 'Update Success'
+    else
+      p 'Update Error'
+    end
+  end
+
+  # Default
+  def screen1
+    p 'Welcome to Human Resources!'
+    p '============================'
+    p '1. Log in'
+    p '2. Register'
+    p '3. Forgot password'
+    p '0. Exit'
+    p '============================'
+    p 'Please choose a function'
+  end
+
+  def screen2
+    p 'Welcome to Human Resources!'
+    p '============================'
+    p '1. Team Management'
+    p '2. Leave management'
+    p '3. Notification management'
+    p '4. Logout'
+    p '5. CSV' if @user['role'] == SessionController::ADMIN
+    p '0. Exit'
+    p '============================'
+  end
+
+  def run
+    @ch = true
+    while @ch
+      screen1
+      function = input
+      clear
+      execute(function) if (0..3).cover? function
+    end
+  end
+
+  def execute2(function)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/atruby-team/internship-18-summer/blob/3c0bf6c6d3420a006a4c5cda03d7111306060898/at-dungnguyen/Final/main.rb#L351"></span>Metrics/CyclomaticComplexity: Cyclomatic complexity for execute2 is too high. [14/10]</td>
    </tr>
  </tbody>
</table>